api/base/parsers.py

```diff
@@ -15,9 +18,37 @@ class JSONAPIParser(JSONParser):
     media_type = 'application/vnd.api+json'
     renderer_class = JSONAPIRenderer
 
+    @staticmethod
+    def flatten_data(resource_object, stream, is_list=False):
```

Comment:
The second parameter is not a bytestream- it has already been parsed into JSON. https://github.com/CenterForOpenScience/osf.io/pull/4353/files#diff-f2bbfa936e68cf857b1230ef89da63aaR53

For readability, consider renaming the second parameter accordingly.